src/test/java/TestUtils.java

```diff
@@ -110,10 +130,26 @@ public static void setSearchServletRequestParameters(HttpServletRequest request,
 
     return analytics;
   }
-  /* * Removes the unique id property from a receipt entity JSON string, leaving only the receipt
+
+  /**
+   * Removes the unique id property from a receipt entity JSON string, leaving only the receipt
    * properties.
    */
   public static String extractProperties(String json) {
     return json.substring(json.indexOf("propertyMap"));
   }
+
+  /**
+   * Checks receipt ids in the returned page all match the ids in the expected page.
+   * @return true if all expectedPage receipt ids are found in returnedPage, else false.
+   */
+  public static boolean checkIdsMatch(ImmutableList<Entity> expectedPage, Receipt[] returnedPage) {
+    for (Entity expectedEntity : expectedPage) {
+      if (!Arrays.stream(returnedPage)
+               .anyMatch(receipt -> receipt.getId() == expectedEntity.getKey().getId())) {
+        return false;
+      }
```

Comment:
The stream could also be mapped to the IDs before calling `anyMatch()`.

Reply:
Can you explain what you mean by that?

Reply:
```
Stream<Long> ids = Arrays.stream(returnedPage).map(Receipt::getId);
for (...) {
  if (!ids.anyMatch(id -> id == expectedEntity.getKey().getId()))
```